src/cli/api/local_api.py

```diff
@@ -36,24 +54,140 @@ def __is_config_valid(self, application: str) -> bool:
+            application: The application name for which the input files need to be copied
+            input_directory: The destination where application files need to be stored
+
+        """
+
+
+    def check_valid_network(self, network: str, app_config: AppConfigType) -> bool:
+        """
+        Check if the network name is a valid network for the Application
+
+        Args:
+            network: The network name
+            app_config: Application Configuration containing the available networks
+
+        Returns:
+            bool: True if the given network name is available in application configuration, False otherwise
+        """
+        return True
+
+    def is_experiment_local(self, path: Path) -> bool:
+        """
+        Check if the experiment at the 'path' is remote or local
+
+        Args:
+            path: The location of the experiment
+
+        Returns:
+            bool: True if the experiment at given path is local, False otherwise
+        """
+        # read experiment.json and check the meta->backend->location attribute
+        return True
+
+    def get_experiment_rounds(self, path: Path) -> int:
+        """
+        Get the number of rounds in an experiment
+
+        Args:
+            path: The location of the experiment
+
+        Returns:
+            int: Value of the number of rounds
+        """
+        # read experiment.json and check the meta->number_of_rounds attribute
+        return 1
+
+    def delete_experiment(self, path: Path) -> None:
         pass
 
+
+    def run_experiment(self, path: Path, block: bool) -> Optional[List[ResultType]]:
+        roundSetManager = RoundSetManager()
+        roundSetManager.prepare_input(path)
+        roundSetManager.process()
+        roundSetManager.terminate()
+        return []
+
     def get_experiment(self, name: str) -> ExperimentType:
         pass
 
-    def get_results(self, name: str) -> List[ResultType]:
-        pass
+    def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+        outputConverter = OutputConverter('log_dir', 'output_dir')
 
-    def validate_experiment(self, path: str) -> bool:
-        pass
+        result_list: List[ResultType] = []
+        output_result: List[ResultType] = []
+
+        total_rounds = self.get_experiment_rounds(path)
+        if all_results:
+            for round_number in range(1, total_rounds + 1):
+                output_result.append(outputConverter.convert(round_number, result_list))
+        else:
+            output_result.append(outputConverter.convert(total_rounds, result_list))
+
+        return output_result
+
+    def validate_experiment(self, path: Path) -> Tuple[bool, str]:
+        experiment_json = path / 'experiment.json'
+        if not experiment_json.is_file():
+            return False, 'File experiment.json not found in the current working directory'
+
+        roundSetManager = RoundSetManager()
+        return roundSetManager.validate_asset(path)
```

Comment:
It is by far not enough to only validate the asset to validate the experiment. Maybe add to a docstrings the validations we want to do here.